abipy/lessons/lesson_helper_functions.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+"""
+reusable functions for lessons
+"""
+
+from __future__ import division, print_function
+
+import sys
+import os
+import shutil
+import abipy.abilab as abilab
+import abipy.data as abidata
+
+
+def help(stream=sys.stdout):
+    """
+    Display the tutorial text.
+    """
+    stream.write(__doc__)
```

Comment:
help access a global variable hence the function must be defined in the
module in which this global variable is defined. For sure there are other
tricks (introspection) that can be used to solve the problem but why? help
is a very small function so I don't see any problem if we put these three
lines of code in the different module

On Wed, Jan 21, 2015 at 2:46 PM, Michiel van Setten <
notifications@github.com> wrote:

> In abipy/lessons/lesson_helper_functions.py
> <https://github.com/gmatteo/abipy/pull/52#discussion_r23299523>:
>
> > +"""
> > +
> > +from __future__ import division, print_function
> > +
> > +import sys
> > +import os
> > +import shutil
> > +import abipy.abilab as abilab
> > +import abipy.data as abidata
> > +
> > +
> > +def help(stream=sys.stdout):
> > +    """
> > +    Display the tutorial text.
> > +    """
> > +    stream.write(__doc__)
>
> self.__doc__ ??
> … <#14b0cbec4dbbca15_>
> __file__.__doc__ ? both not I guess :-( Michiel van Setten UCL, NAPS
> mjvansetten@gmail.com michiel.vansetten@uclouvain.be 010 478824
> '"Although the way may not be obvious at first unless you're Dutch" from
> The Zen of Python
> On 21 Jan 2015, at 14:25, gmatteo wrote: In
> abipy/lessons/lesson_helper_functions.py: > +""" > + > +from __future__
> import division, print_function > + > +import sys > +import os > +import
> shutil > +import abipy.abilab as abilab > +import abipy.data as abidata > +
> > + > +def help(stream=sys.stdout): > + """ > + Display the tutorial text.
> > + """ > + stream.write(__doc__) this won't work because it returns the
> doc string of lesson_helper_functions.py — Reply to this email directly or
> view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gmatteo/abipy/pull/52/files#r23299523>.
>



-- 
Nel 1969, con la potenza di calcolo di due Commodore 64, l'uomo è riuscito
a sbarcare sulla luna. Oggi, per far girare decentemente Windows Vista,
serve 1GB di RAM... qualcosa deve essere andato storto